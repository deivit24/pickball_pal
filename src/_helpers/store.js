import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
