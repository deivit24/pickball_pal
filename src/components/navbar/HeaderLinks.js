import React from 'react';
// react components for routing our app without refresh
import { Link } from 'react-router-dom';
//logout action redux
import { logout } from '../../_actions/auth';
import { useSelector, useDispatch } from 'react-redux';
// @material-ui/core components
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';

// @material-ui/icons

// core components

import Button from './Button';
import styles from '../../assets/js/headerLinksStyle';

const useStyles = makeStyles(styles);

export default function HeaderLinks(props) {
  const auth = useSelector((store) => store.auth);
  const dispatch = useDispatch();
  const classes = useStyles();

  const { isAuth, loading } = auth;
  const handleLogout = async (e) => {
    e.preventDefault();
    dispatch(logout());
  };
  const authLinks = (
    <>
      <ListItem className={classes.listItem}>
        <Link to="/dashboard">
          <Button color="transparent" className={classes.navLink}>
            <i className="fas fa-grip-horizontal"></i> Dashboard
          </Button>
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Link to="/posts">
          <Button color="transparent" className={classes.navLink}>
            <i className="fas fa-comment-alt"></i> Posts
          </Button>
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Button
          onClick={handleLogout}
          color="transparent"
          className={classes.navLink}
        >
          <i className="fas fa-sign-out-alt"></i> Logout
        </Button>
      </ListItem>
    </>
  );

  const guestLinks = (
    <>
      <ListItem className={classes.listItem}>
        <Link to="/login">
          <Button color="transparent" className={classes.navLink}>
            Login
          </Button>
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Link to="/register">
          <Button color="transparent" className={classes.navLink}>
            Register
          </Button>
        </Link>
      </ListItem>
    </>
  );
  return (
    <List className={classes.list}>
      <ListItem className={classes.listItem}>
        <Link to="/profiles">
          <Button color="transparent" className={classes.navLink}>
            <i className="fas fa-users"></i> Players
          </Button>
        </Link>
      </ListItem>
      {!loading && <> {isAuth ? authLinks : guestLinks} </>}
    </List>
  );
}
