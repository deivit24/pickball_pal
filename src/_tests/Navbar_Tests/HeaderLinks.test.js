import React from 'react';
import { render } from '@testing-library/react';
import HeaderLinks from '../../components/navbar/HeaderLinks';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing HeaderLinks

describe('Testing the HeaderLinks', () => {
  it('it renders HeaderLinks without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <HeaderLinks />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <HeaderLinks />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
