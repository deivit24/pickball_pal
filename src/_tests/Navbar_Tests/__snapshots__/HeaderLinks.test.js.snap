// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the HeaderLinks matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root makeStyles-list-35 MuiList-padding"
  >
    <li
      class="MuiListItem-root makeStyles-listItem-36 MuiListItem-gutters"
    >
      <a
        href="/profiles"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-47 makeStyles-transparent-61 makeStyles-navLink-38"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <i
              class="fas fa-users"
            />
             Players
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
